### PR TITLE
Automatically create test results elements for each test suite

### PR DESCRIPTION
**Pull request by jvandort**
Previously, we created a binary test results elements for each test target. When performing aggregation, this only worked if the target test suite had a single target. Otherwise, we'd run into ambiguity.

To resolve this for multi-target test suites, we only create a single variant for each suite and add all test results variants for all targets to the suite's variant. This means when aggregating against a project, we will aggregate the results of all targets.

Eventually we will want to go back to a test results variant for each target, but until we have the targets able to declare their identity via attributes, we will have a single variant per suite.

We remove the concept of a test suite type, as this was a very simple mapping over the suite name, and only differed for the case of the default test suite, which has a test type of UNIT_TEST instead of its suite name. We will eventually want to not use the suite name as an attribute either, but rather use the suite name as part of a capability.

The reports will eventually want to use attributes to distinguish which targets to aggregate instead of aggregating all targets from each suite.

We also make similar changes to Jacoco.

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by jvandort** [comment]
@bot-gradle test this

**Comment by jvandort** [comment]
@bot-gradle test this

**Comment by jvandort** [comment]
@bot-gradle test this

**Comment by jvandort** [comment]
@bot-gradle test this

**Comment by jvandort** [comment]
Not sure what is going on with the docs tests. It seems to be complaining about contents of files that are no longer present in the current version of this PR

**Comment by jvandort** [comment]
@bot-gradle test this

File: platforms/software/test-suites-base/src/main/java/org/gradle/testing/base/plugins/TestSuiteBasePlugin.java [hunk 1]
```diff
@@ -16,9 +16,18 @@
 
 package org.gradle.testing.base.plugins;
 
+import org.apache.commons.lang3.StringUtils;
```
Comment (big-guy) [inline platforms/software/test-suites-base/src/main/java/org/gradle/testing/base/plugins/TestSuiteBasePlugin.java:19]: Can we keep using org.apache.commons.lang.StringUtils? I'm not sure why we have both versions...

**Review by big-guy — APPROVED** [review]
As a follow up, please add something to the upgrade guide that warns about changes in the JaCoCo and test suite plugins + aggregation.

File: platforms/software/test-suites-base/src/main/java/org/gradle/testing/base/plugins/TestSuiteBasePlugin.java (hunk 1 above)
Comment (jvandort) [reply platforms/software/test-suites-base/src/main/java/org/gradle/testing/base/plugins/TestSuiteBasePlugin.java:19]: Shouldn't we prefer lang3 over lang?

Comment (big-guy) [reply platforms/software/test-suites-base/src/main/java/org/gradle/testing/base/plugins/TestSuiteBasePlugin.java:19]: Bah, it looks like we have a mix of both. I thought we were only using lang, not lang3

Comment (big-guy) [reply platforms/software/test-suites-base/src/main/java/org/gradle/testing/base/plugins/TestSuiteBasePlugin.java:19]: I guess lang3 is good enough.  Ignore me

**Comment by jvandort** [comment]
@big-guy  https://github.com/gradle/gradle/pull/31809

**Pull request merged**